bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Display.java

```diff
@@ -2404,6 +2407,12 @@ protected void init () {
 	initFonts ();
 	setDeviceZoom ();
 
+	/*
+	 * If this property exists then use smaller padding for item heights in Table, Tree and List controls
+	 * TODO: this is initialized here, could this be put in a better place? And use a better property name?
+	 */
+	smallItemHeight = System.getProperty("org.eclipse.swt.internal.cocoa.smallItemHeight") != null;
```

Comment:
> Couldn't the setFont/setItemHeight calls be made unconditionally, i.e. even when smallItemHeight was false?

That would be possible, but an overhead if the initial heights are the same as the ones set in `setFont/setItemHeight`

> What if instead introducing a boolean property, we would allow "vertical cell padding" to be configured as an integer? 

I have no personal preference, so that's something that users should decide.

Whatever method is chosen, the important thing is to keep the heights the same when the control is created and when `setFont/setItemHeight` is called, so we don't get "magical shrinkage".

Reply:
> What if instead introducing a boolean property, we would allow "vertical cell padding" to be configured as an integer?

To get the old item height back in Eclipse something like this would need to be added in `eclipse.ini`:

```
-Dorg.eclipse.swt.internal.cocoa.itemPadding=1
```

If this option is not set, the internal default would be 8 to match native macOS padding.